### PR TITLE
New CouchDB Cheat Sheet

### PR DESCRIPTION
**Pull request by pjhampton**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [X] New Instant Answer
    - [X] Cheat Sheets: **`New CouchDB Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

An initial release candidate for the CouchDB cheat sheet.

###### People to notify (@mention interested parties)

cc @GuiltyDolphin @tagawa @moollaza 

cc @duckduckgo/duckduckhack-contributors - Anybody with an interest in CouchDB 1.x here?

---

Instant Answer Page: https://duck.co/ia/view/couchdb_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

screen shot 2016-08-12 at 12 29 53 pm
screen shot 2016-08-12 at 12 29 39 pm

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
+{
+    "id": "couchdb_cheat_sheet",
+    "name": "CouchDB Cheat Sheet",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/couchdb.json]: Just `CouchDB`

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    "description": "An overview of the CouchDB HTTP API",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/couchdb.json]: `CouchDB HTTP API Overview`

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    "metadata": {
+        "sourceName": "CouchDB",
+        "sourceUrl" : "http://docs.couchdb.org/"
+    },
+
+    "template_type": "terminal",
+    
+    "aliases": [
+        "couchdb", "couch", "couch api", "couchdb api"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/couchdb.json]: We can lose the first alias.

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                "key": "http://localhost:5984/_utils",
+                "val": "Starts a CouchDB instance"
+            },
+            {
+                "key": "brew install couchdb",
+                "val": "Installs CouchDB on Mac OSx"
+            },
+            {
+                "key": "sudo apt-get install couchdb -y",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/couchdb.json]: Just `apt-get install...` (never specify sudo - people should know when they do/don't need it).

Comment (pjhampton) [inline share/goodie/cheat_sheets/json/couchdb.json]: Agreed 👍

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=couchdb cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/couchdb_cheat_sheet_mobile.png?xI5Kewd0IH)

**Comment by tagawa** [comment]
Hi @pjhampton. Thanks for another Instant Answer. I just updated the ID/name and Perl module, and now it seems to be happily installed on the beta server for testing: https://beta.duckduckgo.com/?q=couchdb+cheat+sheet&ia=cheatsheet&iax=1

We can leave it there for community review and then merge in a few days if no further feedback.

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                "key": "apt-get install couchdb -y",
+                "val": "Installs CouchDB on Ubuntu"
+            }
+        ],
+
+        "HTTP API": [
+            {
+                "key": "curl -XGET localhost:5984",
+                "val": "retrieves information about this node (GET)"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/couchdb.json]: @pjhampton Lets make the capitalization of first letter in val field consistent

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=couchdb cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/couchdb_cheat_sheet_mobile.png?WpXZTeUkqi)

File: share/goodie/cheat_sheets/json/couchdb.json [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+                "val": "Returns all the documents in the database (GET)"
+            },
+            {
+                "key": "curl -XGET localhost:5984/\\{db_name\\}/\\{doc_id\\}",
+                "val": "Returns the document (GET)"
+            }
+        ]
+
+    }
+}
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/couchdb.json]: Add newline at end of file.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=couchdb cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/couchdb_cheat_sheet_mobile.png?V6v3v2n43v)

**Comment by GuiltyDolphin** [comment]
@pjhampton Thanks! :dolphin:

**Comment by daxtheduck** [comment]
@pjhampton this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**